uaclient/conftest.py

```diff
@@ -98,50 +97,64 @@ def logging_sandbox():
 @pytest.fixture
 def FakeConfig(tmpdir):
     class _FakeConfig(UAConfig):
-        def __init__(self, cfg_overrides={}, features_override=None) -> None:
-            cfg_overrides.update({"data_dir": tmpdir.strpath})
+        def __init__(
+            self, cfg_overrides={}, attached=False, additional_info={}
+        ) -> None:
+            if not cfg_overrides.get("data_dir", None):
+                cfg_overrides.update({"data_dir": tmpdir.strpath})
             super().__init__(cfg_overrides)
-
-        @classmethod
-        def for_attached_machine(
-            cls,
-            account_name: str = "test_account",
-            machine_token: Dict[str, Any] = None,
-            status_cache: Dict[str, Any] = None,
-        ):
-            if not machine_token:
-                machine_token = {
-                    "availableResources": [],
-                    "machineToken": "not-null",
-                    "machineTokenInfo": {
-                        "machineId": "test_machine_id",
-                        "accountInfo": {
-                            "id": "acct-1",
-                            "name": account_name,
-                            "createdAt": "2019-06-14T06:45:50Z",
-                            "externalAccountIDs": [
-                                {"IDs": ["id1"], "Origin": "AWS"}
-                            ],
-                        },
-                        "contractInfo": {
-                            "id": "cid",
-                            "name": "test_contract",
-                            "createdAt": "2020-05-08T19:02:26Z",
-                            "effectiveFrom": "2000-05-08T19:02:26Z",
-                            "effectiveTo": "2040-05-08T19:02:26Z",
-                            "resourceEntitlements": [],
-                            "products": ["free"],
-                        },
+            self._attached = attached
+            if self._attached:
+                account_name = additional_info.get(
+                    "account_name", "test_account"
+                )
+                _machine_token = additional_info.get("machine_token", None)
+                self._status_cache = additional_info.get("status_cache", None)
+                if not _machine_token:
+                    _machine_token = self.default_machine_token(account_name)
+                self.machine_token_file.write(_machine_token)
+                if self._status_cache:
+                    self.write_cache("status-cache", self._status_cache)
+                else:
+                    self.write_cache("status-cache", {"attached": True})
```

Comment:
Sorry, I think I am missing the point of this refactor. Couldn't we just have changed the `for_attached_machine` logic to use `config.machine_token_file.write_cache` ?

Reply:
There was a reason as I was starting out the PR, but I think it got solved as I writing the classes. However, lemme check if it still holds.